### ISSUE TITLE
คือตัวอย่างโค้ด Python สำหรับ AI แบบง่าย ๆ

### ISSUE THREAD

**Issue by Nattawitbell** [Nattawitbell/Nattawit#2]
แน่นอน! นี่คือตัวอย่างโค้ด Python สำหรับ AI แบบง่าย ๆ โดยใช้ TensorFlow และ Keras เพื่อสร้างโมเดลที่สามารถทำนายค่าจากข้อมูลตัวเลขง่าย ๆ

ตัวอย่าง: โมเดล AI ทำนายค่าของฟังก์ชันเชิงเส้น (y = 2x + 1)

import tensorflow as tf import numpy as np # สร้างข้อมูลสำหรับการเทรน X_train = np.array([0, 1, 2, 3, 4, 5], dtype=float) y_train = np.array([1, 3, 5, 7, 9, 11], dtype=float) # y = 2x + 1 # สร้างโมเดล AI (Neural Network แบบง่าย ๆ) model = tf.keras.Sequential([ tf.keras.layers.Dense(units=1, input_shape=[1]) # มี 1 นิวรอนและอินพุต 1 ค่า ]) # คอมไพล์โมเดล model.compile(optimizer='sgd', loss='mean_squared_error') # เทรนโมเดล print("กำลังฝึก AI...") model.fit(X_train, y_train, epochs=500, verbose=0) # ฝึก 500 รอบ # ทดสอบโมเดล X_test = 10 prediction = model.predict([X_test]) print(f"AI ทำนายว่า y = {prediction[0][0]:.2f} เมื่อ x = {X_test}") 

วิธีใช้งาน

ติดตั้ง TensorFlow ก่อน pip install tensorflow 

รันโค้ดด้านบนใน Python

โปรแกรมจะฝึก AI และให้ผลลัพธ์เป็นค่าที่ AI ทำนายได้

โค้ดนี้เป็นโมเดลง่าย ๆ ที่ใช้ Neural Network แบบมีเพียง 1 ชั้นและ 1 นิวรอน โดยโมเดลจะเรียนรู้ฟังก์ชัน y = 2x + 1 จากข้อมูลตัวอย่าง และสามารถนำไปพัฒนาให้ซับซ้อนขึ้นได้
